src/libstd/sync/mutex.rs

```diff
@@ -107,6 +107,61 @@ use crate::sys_common::poison::{self, LockResult, TryLockError, TryLockResult};
 ///
 /// *guard += 1;
 /// ```
+///
+/// It is sometimes a good idea (or even necessary) to manually drop the mutex
```

Comment:
Since this documentation does not go into what makes it a good idea when unnecessary, I think it would be clearer to focus on the necessary situation.

```suggestion
/// It is sometimes necessary to manually drop the mutex
```